### PR TITLE
Fix multiple annotator issues

### PR DESCRIPTION
**Pull request by bartcharbon**
#### Checklist
- [ ] Unit tests
- [ ] Updated testplan

### CODE REVIEW

File: molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/entity/impl/CaddAnnotator.java [hunk 1]
```diff
@@ -58,11 +58,11 @@
 
 	@Autowired
 	private AttributeMetaDataFactory attributeMetaDataFactory;
+
 	private RepositoryAnnotatorImpl annotator;
 
 	@Bean
-	public RepositoryAnnotator cadd()
-	{
+	public RepositoryAnnotator cadd(){
```
Comment (dennishendriksen) [inline molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/entity/impl/CaddAnnotator.java]: Why didn't IntelliJ format this line?

File: molgenis-data-annotators/src/test/java/org/molgenis/data/annotation/resources/impl/ResourceImplTest.java [hunk 2]
```diff
@@ -35,7 +35,7 @@ public void beforeMethod()
 			@Override
 			public RepositoryFactory getRepositoryFactory()
 			{
-				return new TabixVcfRepositoryFactory("cadd");
+				return null;//new TabixVcfRepositoryFactory("cadd");FIXME
```
Comment (dennishendriksen) [inline molgenis-data-annotators/src/test/java/org/molgenis/data/annotation/resources/impl/ResourceImplTest.java]: If this is not supposed to be fixed right now, add to meta data editing to do doc

Comment (bartcharbon) [inline molgenis-data-annotators/src/test/java/org/molgenis/data/annotation/resources/impl/ResourceImplTest.java]: it is... on the very top "fix annotator unit tests"

File: molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/resources/impl/TabixVcfRepositoryFactory.java [hunk 3]
```diff
@@ -14,15 +17,23 @@
...
-	public TabixVcfRepositoryFactory(String name)
+	private VcfAttributes vcfAttributes;
+	private EntityMetaDataFactory entityMetaDataFactory;
+	private AttributeMetaDataFactory attributeMetaDataFactory;
+
+	public TabixVcfRepositoryFactory(String name, VcfAttributes vcfAttributes, EntityMetaDataFactory entityMetaFactory,
+			AttributeMetaDataFactory attrMetaFactory)
 	{
 		this.name = name;
+		this.entityMetaDataFactory = entityMetaFactory;
```
Comment (fdlk) [inline molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/resources/impl/TabixVcfRepositoryFactory.java]: Use requireNonNull()

File: molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/resources/impl/ResourcesImpl.java [hunk 4]
```diff
@@ -25,6 +24,9 @@
 	@Autowired
 	private DataService dataService;
 
+	@Autowired
+	private ApplicationContext applicationContext;
+
```
Comment (fdlk) [inline molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/resources/impl/ResourcesImpl.java]: Nice!

File: molgenis-data-annotators/src/test/java/org/molgenis/data/annotation/entity/impl/GavinAnnotatorTest.java [hunk 5]
```diff
@@ -3,221 +3,253 @@
...
 		@Bean
-		public Entity VariantClassificationAnnotatorSettings()
+		public Entity gavinAnnotatorSettings()
 		{
-			Entity settings = new DynamicEntity(null);
-
-			settings.set(GavinAnnotatorSettings.Meta.VARIANT_FILE_LOCATION,
-					ResourceUtils.getFile(getClass(), "/gavin/GAVIN_calibrations_r0.1.xlsx").getPath());
+			Entity settings = mock(Entity.class);
+			when(settings.getString(GavinAnnotatorSettings.Meta.VARIANT_FILE_LOCATION))
```
Comment (fdlk) [inline molgenis-data-annotators/src/test/java/org/molgenis/data/annotation/entity/impl/GavinAnnotatorTest.java]: Should use typed version of the getter, I think

File: molgenis-data-annotators/src/test/java/org/molgenis/data/annotation/entity/impl/HPOResultFilterTest.java [hunk 6]
```diff
@@ -1,35 +1,107 @@
...
+		EntityMetaData entityMeta = entityMetaDataFactory.create().setSimpleName("HPO");
+		entityMeta.addAttribute(attributeMetaDataFactory.create().setName(HPO_DISEASE_ID_COL_NAME));
+		entityMeta.addAttribute(attributeMetaDataFactory.create().setName(HPO_GENE_SYMBOL_COL_NAME));
+		entityMeta.addAttribute(attributeMetaDataFactory.create().setName(HPO_ID_COL_NAME), ROLE_ID);
+		entityMeta.addAttribute(attributeMetaDataFactory.create().setName(HPO_TERM_COL_NAME));
 
-		Entity e1 = new DynamicEntity(null); // FIXME pass entity meta data instead of null
-		e1.set(HPORepository.HPO_ID_COL_NAME, "id1");
-		e1.set(HPORepository.HPO_TERM_COL_NAME, "term1");
+		Entity e1 = new DynamicEntity(entityMeta); // FIXME pass entity meta data instead of null
```
Comment (fdlk) [inline molgenis-data-annotators/src/test/java/org/molgenis/data/annotation/entity/impl/HPOResultFilterTest.java]: can remove the FIXME

File: molgenis-data-validation/src/main/java/org/molgenis/data/validation/RepositoryValidationDecorator.java [hunk 7]
```diff
@@ -562,7 +562,16 @@ private void validateEntityValueReadOnly(Entity entity, ValidationResource valid
...
-			Object existingValue = entityToUpdate.get(readonlyAttr.getName());
+
+			Object existingValue = null;
+			try
+			{
+				existingValue = entityToUpdate.get(readonlyAttr.getName());
+			}
+			catch (Exception e)
+			{
+				System.out.print("test");
```
Comment (fdlk) [inline molgenis-data-validation/src/main/java/org/molgenis/data/validation/RepositoryValidationDecorator.java]: Not intended, probably? You hid this easter egg well.

File: molgenis-data-import/src/main/java/org/molgenis/data/importer/IntermediateParseResults.java [hunk 8]
```diff
@@ -107,8 +108,7 @@ public EntityMetaData addEntityMetaData(String name)
 			}
 		}
 
-		EntityMetaData emd = new EntityMetaData(simpleName,
-				getApplicationContext().getBean(EntityMetaDataMetaData.class));
+		EntityMetaData emd = new EntityMetaData(simpleName, applicationContext.getBean(EntityMetaDataMetaData.class));
```
Comment (fdlk) [inline molgenis-data-import/src/main/java/org/molgenis/data/importer/IntermediateParseResults.java]: Wouldn't it be better to use entityMetaDataFactory.create() instead and prevent injection of ApplicationContext?

File: molgenis-data/src/main/java/org/molgenis/util/EntityUtils.java [hunk 9]
```diff
@@ -278,7 +279,17 @@ public static boolean equals(EntityMetaData entityMeta, EntityMetaData otherEnti
 		{
 			return false;
 		}
-		if (!entityMeta.getBackend().equals(otherEntityMeta.getBackend()))
+		String backend = entityMeta.getBackend();
```
Comment (fdlk) [inline molgenis-data/src/main/java/org/molgenis/util/EntityUtils.java]: In context, backend null equals the default backend. Bit unsure about this.

**Pull request merged**